core/common/src/main/java/alluxio/util/ShellUtils.java

```diff
@@ -193,6 +194,101 @@ private String run() throws ExitCodeException, IOException {
       }
     }
+
+    /**
+     * Runs a command and returns its stdout, stderr and exit code.
+     * No matter it succeeds or not.
+     *
+     * @return {@link CommandReturn} object representation of stdout, stderr and exit code
+     */
+    protected CommandReturn runTolerateFailure() throws IOException {
+      Process process = new ProcessBuilder(mCommand).start();
+
+      BufferedReader inReader =
+              new BufferedReader(new InputStreamReader(process.getInputStream()));
+      BufferedReader errReader =
+              new BufferedReader(new InputStreamReader(process.getErrorStream()));
+
+      try {
+        // read the output of the command
+        StringBuilder stdout = new StringBuilder();
+        String outLine = inReader.readLine();
+        while (outLine != null) {
+          stdout.append(outLine);
+          stdout.append("\n");
+          outLine = inReader.readLine();
+        }
+
+        // read the stderr of the command
+        StringBuilder stderr = new StringBuilder();
+        String errLine = errReader.readLine();
+        while (errLine != null) {
+          stderr.append(errLine);
+          stderr.append("\n");
+          errLine = inReader.readLine();
+        }
+
+        // wait for the process to finish and check the exit code
+        int exitCode = process.waitFor();
+        if (exitCode != 0) {
+          // log error instead of throwing exception
+          LOG.warn(String.format("Failed to run command %s%nExit Code: %s%nStderr: %s",
+                  Arrays.toString(mCommand), exitCode, stderr.toString()));
+        }
+
+        return new CommandReturn(exitCode, stdout.toString(), stderr.toString());
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        throw new IOException(e);
+      } finally {
+        try {
+          // JDK 7 tries to automatically drain the input streams for us
+          // when the process exits, but since close is not synchronized,
+          // it creates a race if we close the stream first and the same
+          // fd is recycled. the stream draining thread will attempt to
+          // drain that fd!! it may block, OOM, or cause bizarre behavior
+          // see: https://bugs.openjdk.java.net/browse/JDK-8024521
+          // issue is fixed in build 7u60
+          InputStream stdoutStream = process.getInputStream();
+
+          synchronized (stdoutStream) {
+            inReader.close();
+          }
+        } catch (IOException e) {
+          LOG.warn("Error while closing the input stream", e);
+        }
+        try {
+          InputStream stderrStream = process.getErrorStream();
+          synchronized (stderrStream) {
+            errReader.close();
+          }
+        } catch (IOException e) {
+          LOG.warn("Error while closing the error stream", e);
+        }
+        process.destroy();
+      }
+    }
+  }
+
+  @NotThreadSafe
+  private static class SshCommand extends Command {
```

Comment:
This SshCommand automatically does SSH with common parameters. I currently don't see the needs for flexible configuration yet.